libs/shared/src/main/java/io/crate/common/collections/Ordering.java

```diff
@@ -0,0 +1,976 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+
+package io.crate.common.collections;
+
+
+import javax.annotation.Nullable;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.TreeSet;
+
+import static java.util.Objects.requireNonNull;
+
+/*
+ * Extracted from https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/Ordering.java
+ */
+public abstract class Ordering<T> implements Comparator<T> {
```

Comment:
Can we remove some of the unused method? 

We should also follow up later and see how many of the uses we can replace with the `Comparator.` functions

Reply:
I removed already quite a lot of unused stuff. I kept functionality which I thought might be useful in the future. But this can be stripped down much more. I was also thinking of moving everything over to `Comparator`.  I think a follow-up makes sense here.